linkcheck/models.py

```diff
@@ -368,8 +370,13 @@ def check_external(self, external_recheck_interval=EXTERNAL_RECHECK_INTERVAL):
                 else:
                     # Re-raise exception if it's definitely not a false positive
                     raise
+            # If access is denied, possibly the user agent is blocked
+            if response.status_code == HTTPStatus.FORBIDDEN:
+                logger.debug('Forbidden, retry with different user agent')
+                request_params['headers'] = {'User-Agent': FALLBACK_USER_AGENT}
+                response = requests.head(self.external_url, **request_params)
             # If HEAD is not allowed, let's try with GET
-            if response.status_code >= 400:
+            if response.status_code in [HTTPStatus.BAD_REQUEST, HTTPStatus.METHOD_NOT_ALLOWED]:
                 logger.debug('HEAD is not allowed, retry with GET')
                 response = requests.get(self.external_url, **request_params)
```

Comment:
I'm unsure about the order of these checks.
If a server has both limitations (blocking the default user agent and blocking `HEAD` requests), it depends on the server's response which one we would have to handle first...
My guess would be that the HTTP method is checked before the user agent, so in theory we would have to put this case first and then remember whether we have to do `head()` or `get()` with our fallback user agent? Or should we just assume `get()` in any case, because there aren't that many domains where we need the fallback user agent? :thinking: 

Also, do we have to remember the domains which require a different user agent? If we don't (and have a lot of links to this domain), we might do a lot of unnecessary requests?